src/classes/STG_PanelHealthCheck_CTRL.cls

```diff
@@ -673,5 +674,47 @@ public without sharing class STG_PanelHealthCheck_CTRL extends STG_Panel {
         }    
         
     }
-    
+
+    /**
+     * @description Detect if any opportunities exists where the same contact
+     * is associated with that opportunity multiple times through multiple
+     * opportunity contact role records.
+     */
+     public void verifyNoDuplicateContactRoles() {
+        try {
+            List<AggregateResult> results = [
+                SELECT
+                    COUNT(Id),
+                    OpportunityId,
+                    ContactId
+                FROM OpportunityContactRole
+                GROUP BY OpportunityId, ContactId
+                HAVING Count(Id) > 1
```

Comment:
I coded this straight to what the rollup code checks.  There is no check against role being 'Soft Credit' or any specific role at all in the rollup code, where the deduplication takes place.

Yes, calling code may decide to filter by soft credit roles, but the rollup code does not make that distinction, so there is no guarantee that the duplicate counting issue is limited to soft credit roles.